### PR TITLE
[ENG-361] Remove dead prereg code

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

We have completed the prereg challenge! With new reviewer changes being handled on v2 we don't need the old prereg code, so time to delete the old v1 code.

# Changes

- Remove special casing for permissions on prereg for views
- Remove special casing for checking out a prereg file
- Remove special casing for checking out a prereg file permission
- Remove admin `pre_reg` app
- Remove prereg challenge JSON schema
- Remove prereg landing page waffling
- Remove prereg analytics from admin dashboard
- Remove Prereg Challenge specific campaign code (while keeping OSF Preregistration)

## QA Notes

Check out admin app and draft registration process to ensure there's no new bugs. I am most worried about difficult/usually draft registrations using prereg features.

## Documentation

Not user facing

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-361

### CODE REVIEW

File: admin/base/settings/defaults.py [hunk 1]
```diff
@@ -233,9 +232,6 @@
     })
 
 
-ENTRY_POINTS = {'osf4m': 'osf4m', 'prereg_challenge_campaign': 'prereg',
```
Comment (Johnetordoff) [inline admin/base/settings/defaults.py]: [this line was made obsolete when this line was removed](https://github.com/Johnetordoff/osf.io/commit/3a7e6ea06e76645182427ca94c5eddd4f02c71cc#diff-76a740757630125060c6bdefa75efb58L8)

File: admin/base/settings/defaults.py [hunk 2]
```diff
@@ -249,7 +245,7 @@
     INSTALLED_APPS += ('debug_toolbar', 'nplusone.ext.django',)
     MIDDLEWARE += ('debug_toolbar.middleware.DebugToolbarMiddleware', 'nplusone.ext.django.NPlusOneMiddleware',)
     DEBUG_TOOLBAR_CONFIG = {
-        'SHOW_TOOLBAR_CALLBACK': lambda(_): True,
+        'SHOW_TOOLBAR_CALLBACK': lambda _: True,
```
Comment (Johnetordoff) [inline admin/base/settings/defaults.py]: Drive-by: just fixing this for pre-commit hooks.

File: osf/models/registrations.py [hunk 3]
```diff
@@ -617,16 +617,6 @@ def update_metadata(self, metadata):
         self.registration_metadata.update(metadata)
         return changes
 
-    def submit_for_review(self, initiated_by, meta, save=False):
```
Comment (Johnetordoff) [inline osf/models/registrations.py]: This is only ever used in tests, so I'm just deleting it.

File: tests/test_registrations/test_views.py [hunk 4]
```diff
@@ -432,15 +370,6 @@ def test_check_draft_state_registered_but_deleted(self):
         except Exception:
             self.fail()
 
-    def test_check_draft_state_pending_review(self):
```
Comment (Johnetordoff) [inline tests/test_registrations/test_views.py]: removed because the function it's testing `submit_for_review` isn't being used.

File: website/project/metadata/test-schema.json [hunk 5]
```diff
@@ -0,0 +1,79 @@
+{
```
Comment (Johnetordoff) [inline website/project/metadata/test-schema.json]: Since we had to move away from using the `prereg-prize.json` as a test value I put in a generic test schema

File: admin/static/js/sales_analytics/dashboard.js [hunk 6]
```diff
@@ -387,19 +381,13 @@ var SalesAnalytics = function() {
                     userProductMap.osf.push(session['user.id']);
                     var paths = session.result;
                     var numberOfProducts = 0;
-                    var meetings, prereg, institutions;
```
Comment (UdayVarkhedkar) [inline admin/static/js/sales_analytics/dashboard.js]: Should `meetings` and `institutions` still be instantiated here? I would expect that `prereg` should be removed from this line rather than the line itself being deleted.

File: framework/auth/campaigns.py [hunk 7]
```diff
@@ -24,7 +24,7 @@ def get_campaigns():
             # Native campaigns: PREREG and ERPC
             newest_campaigns = {
                 'prereg': {
-                    'system_tag': 'prereg_challenge_campaign',
+                    'system_tag': 'osf_prereg_campaign',
```
Comment (UdayVarkhedkar) [inline framework/auth/campaigns.py]: What determined the new system_tag value?

File: api_tests/schemas/views/test_registration_schemas_detail.py [hunk 8]
```diff
@@ -51,9 +51,9 @@ def test_schemas_detail_visibility(self, app, user, schema):
         res = app.get(url, auth=user.auth)
         assert res.status_code == 200
         data = res.json['data']['attributes']
-        assert data['name'] == 'Prereg Challenge'
+        assert data['name'] == 'OSF Preregistration'
         assert data['schema_version'] == 2
-        assert data['active']
+        assert not data['active']
```
Comment (UdayVarkhedkar) [inline api_tests/schemas/views/test_registration_schemas_detail.py]: I can see that this was changed so that the test passes but why did it need to be changed for the test to pass?

File: framework/auth/campaigns.py (hunk 7 above)
Comment (Johnetordoff) [reply framework/auth/campaigns.py]: Me! No actually you have a good point. I was just trying to remove the word `challenge` because it's not a challenge anymore. We could continue to use it internally without ill effect, but I thought lets make the end of the prereg_challenge_campaign very obvious.

File: api_tests/schemas/views/test_registration_schemas_detail.py (hunk 8 above)
Comment (Johnetordoff) [reply api_tests/schemas/views/test_registration_schemas_detail.py]: The ensure_schema function that populates schema data for tests, isn't updated when we activate a schema, so it's out of date and is populating most things as inactive. Instead of reversing this I'm just going to remove this check because it isn't directly relevant to the test.

**Comment by pattisdr** [comment]
hey @Johnetordoff, starting to review now, could you get this up-to-date for me?

**Comment by pattisdr** [comment]
✌️ appreciate it, @Johnetordoff

File: api_tests/wikis/views/test_wiki_detail.py [hunk 9]
```diff
@@ -28,7 +28,7 @@
 
 
 def make_rename_payload(wiki_page):
-    new_page_name = fake.word()
+    new_page_name = 'any acsii string, except "home" sdaww\n\neq'
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply api_tests/wikis/views/test_wiki_detail.py]: There are six instances of this and 970 "random" words so the odds are 6/970 == .6%

File: osf_tests/test_sanctions.py [hunk 10]
```diff
@@ -70,128 +66,3 @@ def test_terminate_embargo_log_is_nouser(self, node, user, registration):
...
-
-@pytest.mark.django_db
-@pytest.mark.enable_quickfiles_creation
-class TestDraftRegistrationApprovals:
-
-    @mock.patch('framework.celery_tasks.handlers.enqueue_task')
-    def test_on_complete_immediate_creates_registration_for_draft_initiator(self, mock_enquque):
-        user = factories.UserFactory()
-        project = factories.ProjectFactory(creator=user)
-        registration_schema = RegistrationSchema.objects.get(name='Prereg Challenge', schema_version=2)
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply osf_tests/test_sanctions.py]: Some of these must be removed because they test the admin permission which was removed, but the rest is reverted.

**Comment by Johnetordoff** [comment]
@pattisdr these travis tests have passed my fork https://travis-ci.org/Johnetordoff/osf.io/builds/588484369

File: osf/migrations/0201_remove_prereg_waffling.py [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('osf', '0200_auto_20200214_1518'),
+    ]
+
+    operations = [
+        DeleteWaffleFlags([features.OSF_PREREGISTRATION]),
```
Comment (aaxelb) [inline osf/migrations/0201_remove_prereg_waffling.py:17]: hm, this migration (and 0144) is the only thing keeping us from deleting `features.OSF_PREREGISTRATION`, and we won't want migrations for adding/removing waffles once #9308 is merged -- wondering if we should remove the migration and the switch now, knowing it'll eventually get cleaned up... thoughts?

Comment (Johnetordoff) [reply osf/migrations/0201_remove_prereg_waffling.py]: Yeah, I want to wait to fix things up, it's likely all these migrations are going to be squashed soon anyway. I'd want to wait unitl #9308 is merged.

**Comment by Johnetordoff** [comment]
merged into staging and tested

**Comment by brianjgeiger** [comment]
Looks like there's a migration that needs updating. Or you could remove it, as we'll be handling those flags in a better way soon.

**Comment by Johnetordoff** [comment]
@brianjgeiger I removed the migration, but you may have to clear cache for testmon, migration fails tend to stick in the cache because they stop the build from completing.
